data_analysis/correction_framework.py

```diff
@@ -0,0 +1,290 @@
+        #                          2023-01-23T06:00:00.992 on file
+        self.night_of = night_date
+        self.morning_of = morning_date
+
+        sensor = Observer(
+            longitude=sensor_long * u.deg,
+            latitude=sensor_lat * u.deg,
+            elevation=0 * u.m,
+            name="Sensor",
+            timezone="US/Central",
+        )
+        # reports time in UTC
+        self.sunset = sensor.sun_set_time(night_date, which="nearest")
+        self.sunrise = sensor.sun_rise_time(morning_date, which="nearest")
+        self.astronomical_twilight = sensor.twilight_evening_astronomical(
+            night_date, which="nearest"
+        )
+
+        # how much of the lunar surface is illuminated that night, pulled when
+        # the sky is fully dark (astro. twilight) comparable to moon phase
+        # (0 = new moon, ~1 = full) without the pi of obs.moon_phase
+        self.moon_illumination = sensor.moon_illumination(
+            self.astronomical_twilight
+        )
+        # future TODO: store nightly weather
+
+        self.min_records = []
+
+        # example line from 2023-01-23_LENSSTSL0008.txt:
+        # 2023-01-23T06:00:00.992;2023-01-23T00:00:00.992;0.38;4.39;0.0;LENSS_TSL_0008
+        # UTC Time; Local Time; Temperature (C); Frequency; Voltage; ID
+        with open(night_file) as n_file:
+            trans_str = n_file.readline()
+            while len(trans_str) != 0:
+                data = trans_str.split(";")
+                data[0] = data[0].replace("T", " ")
+                data[1] = data[1].replace("T", " ")
+                m_rec = MinuteRecord(
+                    Time(data[0]),
+                    Time(data[1]),
+                    float(data[2]),
+                    float(data[3]),
+                    float(data[4]),
+                    data[5],
+                )
+                if m_rec.utc_time < self.sunset:
+                    trans_str = n_file.readline()
+                    continue
+                self.min_records.append(m_rec)
+                trans_str = n_file.readline()
+
+        with open(morning_file) as m_file:
+            trans_str = m_file.readline()
+            while len(trans_str) != 0:
+                data = trans_str.split(";")
+                data[0] = data[0].replace("T", " ")
+                data[1] = data[1].replace("T", " ")
+                m_rec = MinuteRecord(
+                    Time(data[0]),
+                    Time(data[1]),
+                    float(data[2]),
+                    float(data[3]),
+                    float(data[4]),
+                    data[5],
+                )
+                if m_rec.utc_time > self.sunrise:
+                    break
+                self.min_records.append(m_rec)
+                trans_str = m_file.readline()
+
+        self.valid_for_use = True
+        self.exclusion_reason = "None"
+
+    # grade quality of as based on various night conditions
+    def grade_data(self):
```

Comment:
You have these comments above your functions/methods, but they should be in the standard docstring format.
```python
def grade_data(self):
"""
Grades the quality of data as based on various night conditions.
"""
```